### PR TITLE
Miscellaneous updates

### PR DESCRIPTION
**Pull request by dmichaels-harvard**
* Refactored to use dcicutils.portal_utils.Portal.
* Many minor-ish changes to submit-metadata-bundle, resume-uploads, upload-item-data.
  E.g. sanity checking file paths and uuids, providing more info/feedback to user,
  allowing accession ID or accession ID based file name, show file sizes, etc.
* Subsume upload-item-data functionality into resume-uploads for convenience.
* Starting (readthedocs) documentation updates.

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -45,9 +45,11 @@ This is a tool for uploading files to SMaHT.
...
 ===================
 
-* Python 3.7, 3.8 or 3.9
-* Pip (>=20.0.0)
-* Virtualenv (>=16.0.0)
+* python (>=3.8 and <=3.11) 
+* pip (>=22.0.0)
+* poetry (>=1.4.0)
+* pyenv (>=2.3.0)
+* smaht-portal (>=1.0.0)
```
Comment (willronchetti) [inline docs/source/index.rst:52]: This is not strictly speaking true no? This is only if you want to develop it locally, not sure we want to say so if outside users are seeing this

File: docs/source/installation.rst [hunk 2]
```diff
@@ -90,33 +91,37 @@ just do::
...
+           "key": "your-portal-key",
+           "secret": "your-portal-secret",
            "server": "https://data.smaht.org"
        },
        "smaht-local": {
-           "key": "some_other_key",
-           "secret": "some_other_secret",
+           "key": "your-local-portal-key",
+           "secret": "your-local-portal-secret",
            "server": "http://localhost:8000"
```
Comment (willronchetti) [inline docs/source/installation.rst:110]: Again I'm hesitant to include such info in what may be user facing documentation, you might consider moving this into a "developer tricks" file or something of a sort just so it's clear it's not the intended use for an actual user

**Review by willronchetti — APPROVED** [review]
Looks good just some small comments

File: docs/source/index.rst (hunk 1 above)
Comment (dmichaels-harvard) [reply docs/source/index.rst:52]: Not sure which part you're saying is not strictly true - you mean the version numbers or the inclusion of pip or poetry etc?

File: docs/source/installation.rst (hunk 2 above)
Comment (dmichaels-harvard) [reply docs/source/installation.rst:110]: I assume you mean just the "smaht-local" part.

**Pull request merged**